packages/nimble-components/src/testing/async-helpers.ts

```diff
@@ -11,3 +11,10 @@ export const waitForUpdatesAsync = async (): Promise<void> => DOM.nextUpdate();
  * updates that resulted from DOM interactions like property or attribute changes.
  */
 export const processUpdates = (): void => DOM.processUpdates();
+
+/**
+ * Waits for a requested animation frame to occur.
+ */
+export const waitAnimationFrame = async (): Promise<void> => new Promise(resolve => {
+    requestAnimationFrame(() => resolve());
+});
```

Comment:
lets not put this in testing since that's a public api and this is for a very specific case. Instead could be in page objects or /patterns/tests or somewhere under the root /tests